File: Stack/src/Stack.java
public class Stack {
	private static class Node {
		private String element;
		private Node next;
		
		// Constructor
		public Node (String e, Node n) {
			element = e;
			next = n;
		}
		
		public String getElement() {
			return element;
		}
		
		public Node getNext() {
			return next;
		}
		
		public void setNext(Node n) {
			next = n;
		}
	}
	
	private Node head = null;
	private Node tail = null;
	int size = 0;
	public Stack() {}
	
	public int size() {
		return size;
	}
	
	public boolean isEmpty() {
		return size == 0;
	}
	
	public String first() {
		if (isEmpty()) {
			return null;
		}
		return head.getElement();
	}
	
	public String last() {
		if (isEmpty()) {
			return null;
		}
		return tail.getElement();
	}
	
	public void push(String e) {
		Node newest = new Node(e, head);
		if (isEmpty()) {
			head = tail = newest;
		} else {
			Node newNext = newest.getNext(); 
			newNext = head;
			head = newest;
		}
		size++;
		return;
	}
	
	public void pop() {
		if (isEmpty()) {
			return;
		}
		head = head.getNext();
		size--;
		if (size == 0) {
			tail = null;
		}
	}

}

File: Stack/Stack.cpp
```
#include <iostream>
#include "Stack.h"

class Node 
{
private:
	int nData;
	Node* pNext;
public:		
		// Constructor
	
};
Node::Node (int nData) 
{
	this->nData = nData;
	this->pNext = nullptr;
}
		
int Node::GetData() 
{
	return nData;
}
Node* Node::GetNext()
{
    return pNext;
}

// Default constructor
Stack::Stack() 
{
	//
}
	
int Stack::Size() 
{
	return nSize;
}

bool Stack::IsEmpty() 
{
	return (nSize == 0);
}

int Stack::First() 
{
	if (IsEmpty()) 
	{
	    std::cout << "List is empty!\n";
		return 0;
	}
	return pHead->GetData();
}

int Stack::Last() 
{
	if (IsEmpty()) 
	{
		std::cout << "List is empty!\n";
		return 0;
	}
	return pTail->GetData();
}

void Stack::Push(int nData) 
{
	Node* pNewNode = new Node(nData);
	if (IsEmpty()) 
	{
		pHead = pTail = pNewNode;
	} 
	else 
	{
	    Node* pNextNode = pNewNode->GetNext();
		pNextNode = pHead;
		pHead = pNewNode;
	}
	nSize++;
	return;
}

void Stack::Pop() 
{
	if (IsEmpty()) 
	{
		return;
	}
	pHead = pHead->GetNext();
	nSize--;
	if (nSize == 0) 
	{
		pTail = nullptr;
	}
}

```
